### DISCUSSION TITLE
Ability to construct the graph through text files

### DISCUSSION THREAD

**Discussion by GrimPixel** [IndiePython/nodezator#75] [Feature Request]
The connections between nodes can be specified in a TSV file.

Nah, I think the current one is good. Using TSV requires multiple files to define a graph.

**Comment by KennedyRichard** [comment]
Oh, I see you gave up on your original idea after second thought and already closed the discussion yourself.

Considering/brainstorming new ideas and re-examining or discarding them without hesitation is important to achieve high quality ideas in the end, after carefully consideration of all the available options.

Thanks for the attempt anyway, and feel free to keep the ideas coming :+1:

**Comment by GrimPixel** [comment]
OK, let's do it again.

I think there can be a new format for the current one, omitting unimportant information like positions.

File of node definition `node.tsv`:
```
id_a	node_a
id_b	node_b
```
File of edge definition `edge.tsv`:
```
id_a	port_a	id_b	port_b
```


Another way is placing them in one file `graph.tsv`:
```
id_a	node_a		
id_b	node_b		
id_a	port_a	id_b	port_b
```
The program then distinguishes node and edge definitions by checking if the last two fields are empty.

The program then arranges the nodes in an optimised way, like that feature in [Heimer](https://github.com/juzzlin/Heimer).

**Comment by KennedyRichard** [comment]
Noted.

Since there's nothing wrong with how Nodezator manages connections currently, this is probably a feature idea of low priority, that is, something that will take some time until we revisit it. It is interesting though, specially the ability to optimize the layout.
